test/ash_ai/exposed_tools_test.exs

```diff
@@ -0,0 +1,224 @@
+# SPDX-FileCopyrightText: 2024 ash_ai contributors <https://github.com/ash-project/ash_ai/graphs.contributors>
+#
+# SPDX-License-Identifier: MIT
+
+defmodule AshAi.ExposedToolsTest do
+  @moduledoc """
+  Tests for AshAi.exposed_tools/1 filtering logic.
+
+  This tests the core filtering functionality separate from the LLM integration.
+  """
+  use AshAi.RepoCase, async: true
+
+  alias AshAi.Test.Music
+
+  @opts [otp_app: :ash_ai]
+
+  describe "tools filtering" do
+    test "tools: nil returns all tools (default)" do
+      tools = AshAi.exposed_tools(@opts)
+      assert length(tools) == 5
+    end
+
+    test "tools: specific list filters to only those tools" do
+      opts = Keyword.put(@opts, :tools, [:list_artists, :create_artist_after])
+      tools = AshAi.exposed_tools(opts)
+
+      assert length(tools) == 2
+      tool_names = Enum.map(tools, & &1.name)
+      assert :list_artists in tool_names
+      assert :create_artist_after in tool_names
+      refute :update_artist_after in tool_names
+    end
+
+    test "tools: single tool in list" do
+      opts = Keyword.put(@opts, :tools, [:create_artist_manual])
+      tools = AshAi.exposed_tools(opts)
+
+      assert length(tools) == 1
+      assert hd(tools).name == :create_artist_manual
```

Comment:
Personally I would prefer pattern matching here. 

```suggestion
      assert [%{name: :create_artist_manual}]  = AshAi.exposed_tools(opts)
```

I would do that for most of the cases here, at least for the lists that have < 3 elements